src/parlant/api/sessions.py

```diff
@@ -1307,7 +1337,12 @@ async def _add_agent_message(
 
         session = await session_store.read_session(session_id)
 
-        correlation_id = await application.dispatch_processing_task(session)
+        if params.actions:
+            actions = [utterance_request_dto_to_utterance_request(a) for a in params.actions]
+            correlation_id = await application.dispatch_utter_task(session, actions)
```

Comment:
Didn't we discuss that the API should return only when the uttering is completed? So that the tool would not be in a rate with the uttering.

If so then why are we dispatching a background task and returning immediately?